app/javascript/packages/validated-field/validated-field.tsx

```diff
@@ -20,6 +28,11 @@ interface ValidatedFieldProps {
    */
   validate?: ValidatedFieldValidator;
 
+  /**
+   * Optional key and value that indicates the error and resulting error message
+   */
+  messages?: Object;
```

Comment:
should we make this a record? (it looks like the `getErrorMessages` returns a slightly more complex type than this but it's the gist)
```suggestion
  messages?: Record<string, string>;
```

Reply:
The `Object` type is also discouraged to be used ([related docs](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#general-types)).

To @zachmargolis's point, I tend to prefer `Record` over `object` since you can use it to define the types of values you're expecting in the object (strings in this case).